src/conf_mode/interfaces-wireguard.py

```diff
@@ -105,6 +107,9 @@ def verify(wireguard):
         if peer['public_key'] in public_keys:
             raise ConfigError(f'Duplicate public-key defined on peer "{tmp}"')
 
+        if 'disable' not in peer and is_wireguard_key_pair(wireguard['private_key'], peer['public_key']):
+            raise ConfigError(f'Peer "{tmp}" has the same public key as the router')
```

Comment:
… as the router. This sounds weird in my opinion. It should state that it is the public key portion of the interfaces private key to make it more obvious.

Reply:
Done. 
`Peer "TEST" has the same public key as the interface "wg1"`
And I have changed the migration script as we discussed. I disable it if it satisfies the conditions.
I have created the smoketest.